apps/mitt-konto/src/MittKonto/Main/Views/UserView/Subscription/Renew.purs

```diff
@@ -0,0 +1,329 @@
+              Right OrderStarted ->
+                [ DOM.div
+                    { className: "payment-wrapper"
+                    , children:
+                        [ DOM.div
+                            { className: "payment-terminal"
+                            , children: [ DOM.text "Vänligen vänta. Vi behandlar din beställning." ]
+                            }
+                        ]
+                    }
+                ]
+              _ -> mempty
+            _ -> mempty
+        )
+
+renderForm :: RenewSubscription -> State -> ((State -> State) -> Effect Unit) -> Form -> JSX
+renderForm props state setState form =
+  DOM.div
+    { className: "clearfix renew-subscription--container"
+    , children:
+        [ Grid.row_
+            [ DOM.div
+                { className: "col col-11"
+                , children: [ DOM.h3_ [ DOM.text "Förnya prenumerationen" ] ]
+                }
+            , DOM.div
+                { className: "col-1 flex renew-subscription--close-icon"
+                , children: [ DOM.div { className: "close-icon" } ]
+                , onClick: handler_ props.onCancel
+                }
+            ]
+        , DOM.form
+            { onSubmit: handler preventDefault $ const submit
+            , children:
+                [ selectPayment
+                , selectOffer
+                , if form.paymentMethod == PaperInvoice then paperInvoiceMessage else mempty
+                , showPrice
+                , if isJust form.confirm then confirmBuy else mempty
+                , buyButton
+                ]
+            }
+        ]
+    }
+  where
+    subsnoStr = Subsno.toString props.subscription.subsno
+    setFormState = \f -> setState $ \s -> s { form = map f s.form }
+    submit = if form.confirm == Just true || form.paymentMethod == CreditCard
+             then submitForm props form state setState
+             else setFormState _ { confirm = Just false }
+
+    selectPayment =
+      DOM.div
+        { className: "input-field--container"
+        , style: DOM.css { "width": "100%" }
+        , children:
+            [ inputLabel { label: "Betalningssätt", nameFor: "sub-po-" <> subsnoStr }
+            , DOM.select
+                { id: "sub-po-" <> subsnoStr
+                , children:
+                    [ DOM.option { value: "c", children: [ DOM.text "Kreditkort" ] }
+                    ] <>
+                    ( if props.hasAddress
+                        then [ DOM.option { value: "i", children: [ DOM.text "Pappersfaktura" ] } ]
+                        else mempty
+                    )
+                , onChange: handler targetValue
+                    (\newMethod -> do
+                        let method = decodeMethod newMethod
+                        setFormState _ { paymentMethod = method
+                                       , confirm = Nothing
+                                       , shownPrice = form.cents + if method == PaperInvoice then 500 else 0
```

Comment:
My mind would be more in peace if we stored the wild `500` into a variable like `paperInvoiceFeeCents` 😁 Especially when we're using it in two places.